---
Title: "P8_project_analysis_pipeline"
Author: TengMa
Time: 2020.08.01
---

## Bioinformatics software and key codes for metagenomic sequencing analysis
# For quality control and de_hosting: KneadData and Bowtie2 
klab_metaqc qc -s data.list -t human -j 10 -o qc_result_folder_rmhuman -f # klab_metaqc is a simple process combining the above two tools

# For assembling: Megahit
ls -d *| parallel -j 5 megahit -m 0.5 --min-contig-len 200 -t 10 --out-dir {}_Output_ass --out-prefix {} -1 {}/{}.rmhost.r1.fq.gz -2 {}/{}.rmhost.r2.fq.gz ::: * # parallel code

# For genome Binning: MaxBin2 and MetaBAT2
ls -d Sample* | parallel -j 3 metawrap binning -a Assembly_result/{}_Output_ass/{}_Output_contigs_min2000.fasta -o {}/{}.bins -t 16 --metabat2 --maxbin2 --concoct -l 2000 {}/{}.rmhost_1.fastq {}/{}.rmhost_2.fastq # parallel code

# For bin_refinement: MetaWRAP
ls -d *bin | parallel -j 5 metawrap bin_refinement -t 5 -m 200 -c 50 -x 10 -A {}/maxbin2_bins -B {}/metabat2_bins -o {}/metawrap # parallel code

# For genome de_replicate: dRep
cat bins_80_5_checkm.summary | awk 'BEGIN{OFS=",";}{print $1 ".fa", $2, $3}' > ./bins_80_5_checkm_info
sed 's/bin.fa,/genome,/g' bins_80_5_checkm_info > bins_80_5_checkm_info.csv
dRep dereplicate dereplicate_result -pa 0.95 -sa 0.95 -g ./bins_80_5/* -p 32 --genomeInfo bins_80_5_checkm_info.csv

# For genome quality evaluate: checkM
checkm lineage_wf -x fa -t 32 --pplacer_threads 32 bins_80_5 bins_80_5_checkm
summarize_checkm.py bins_80_5_checkm/storage/bin_stats_ext.tsv > bins_80_5_checkm.summary

# For gene predicting: Prodigal
ls -d S* | parallel -j 4 bwa index -a bwtsw {}/{}_contigs_min2000.fasta
ls -d S* | parallel -j 4 bwa mem -t 10 {}/{}_Output_contigs_min2000.fasta ../../0.1_clean_data/all_sample/{}/{}.rmhost_1.fastq.gz ../../0.1_clean_data/all_sample/{}/{}.rmhost_2.fastq.gz -o {}/{}.sam 
for i in `ls -d S*`; do samtools view -bS -@ 12 ${i}/${i}.sam | samtools sort - -o ${i}/${i}.sort.bam; done 

# For genome protein annotation: diamond(blastp)
diamond blastp --threads 32 --max-target-seqs 10 --db  /nvmessdnode3/opt/database/uniport/uniprot_trembl_sport.dmnd --query all_combine.faa --outfmt 6 qseqid sseqid stitle pident qlen slen length mismatch gapopen qstart qend sstart send evalue bitscore --out all_combine.dia

# For map raw reads to the scaffolds: BBMap
cat sample_names | parallel -j 10 bbmap.sh ref=all_rename_bins.fa in1=./all_sample/{}/{}_paired_1.fastq.gz in2=./all_sample/{}/{}_paired_2.fastq.gz -Xmx50g threads=10 unpigz=T out=bbmap_out/{}.sam minid=0.95 idfilter=0.95

# For identify of neuroactive compounds
java -jar -Xmx500g /nvmessdnode3/opt/software/omixer/omixer-rpm-1.1/omixer-rpm-1.1.jar -d /nvmessdnode3/opt/software/omixer/omixer-rpm-1.1/new_pipeline_452/new_pathway.f  -i bin_kegg_matrix -t 20 -o omixer_out  -e 2 -c 0.66

## The R code used in present study
# NMDS plot
library(vegan)
library(ggplot2)
setwd("D:\Users\Administrator\Documents\Desktop")
bray <- read.table("bray.txt", row.names = 1, sep = '\t', header = T)
group <- read.table("group.txt", sep = '\t', header = T)
nmds1 <- metaMDS(otu, distance = 'bray', k = 2)
sample_site <- data.frame(nmds1$point)
sample_site$names <- rownames(sample_site)
names(sample_site)[1:2] <- c('NMDS1', 'NMDS2')
# nmds1.species <- data.frame(nmds1$species) 
# nmds1.stress <- nmds1$stress 
sample_site <- merge(sample_site, group, by = 'names', all.x = TRUE)
ggplot(sample_site, aes(NMDS1, NMDS2, group = group)) + geom_point(aes(color = group, shape = group), size = 3.5, alpha = 0.8) + 
	scale_shape_manual(values = c(17, 16, 15)) + 
	scale_color_manual(values = c('red', 'blue', "green")) + 
		theme(panel.grid = element_blank(), panel.background = element_rect(color = 'black', fill = 'transparent')) +
		theme(legend.key = element_rect(fill = 'transparent'), legend.title = element_blank()) + 
			labs(x = 'NMDS axis1', y = 'NMDS axis2', title = paste('Stress =', round(nmds1$stress, 4))) + theme(plot.title = element_text(hjust = 0.5))

# barplot & violinplot 
library(ggpubr)
setwd("D:\Users\Administrator\Documents\Desktop")
data <- read.table("Bray_curtics.txt",sep='\t', header=T, row.names=1, check=F, comment='')
p <- ggboxplot(data, x="group1", y="dat", color = "group1", 
               palette = c("#00AFBB", "#E7B800"), 
               add = "jitter", shape="group1")
g <- ggviolin(data, x="group2", y="dat", fill = "group2", 
         palette = c("#00AFBB", "#E7B800", "#FC4E07"), 
         add = "boxplot", add.params = list(fill="white")) + stat_compare_means(comparisons = my_comparisons, label = "p.signif")

# procrustes analysis
library(reshape2)
library(ggplot2)
library(dplyr)
library(phyloseq)
library(ggsci)
library(ggpubr)
library(plyr)
library(ggpubr)
library(cowplot)
library(vegan)
library(ggcor)
library(ggcorrplot)

mp = read.table("mp.txt", sep="\t", header=T, row.names=1)
# index
index = read.table("index.txt", sep="\t", header=T, row.names=1)
index_bray = vegdist(t(index[, 2:ncol(index)]), method = "bray")
INA = monoMDS(index_bray)$point %>% as.data.frame()
# taxa
taxa_ab = read.table("taxa", sep="\t", header=T, row.names=1)
taxa_ab_bray = vegdist(t(taxa_ab[, 2:ncol(index)]), method = "bray")
INB = monoMDS(taxa_ab_bray)$point %>% as.data.frame()
# order
INA = INA[order(rownames(INA)), ] ; 
INB = INB[order(rownames(INB)), ] ;
# procrustes
procrustes.results <- ade4::procuste(INA, INB)
A = procrustes.results$tabX %>% as.data.frame() ; A$Sample = rownames(A)
B = procrustes.results$tabY %>% as.data.frame() ; B$Sample = rownames(B)
INA_sp = INA;  INA_sp$Sample = rownames(INA)
INB_sp = INB;  INB_sp$Sample = rownames(INB)

plot_pro$method = rep(c("index", "taxa"), each = 314) 
plot_pro = merge(plot_pro, mp, by = "Sample")

###P values
pro_test = protest(X = INA, Y = INB, scores = "sites", permutations = 999)
plot_pro_ggpt = ggplot(plot_pro, aes(MDS1, MDS2, color = Time, shape = method)) + 
    geom_point(size = 3) +
    geom_line(aes(group = Sample, color = Time), alpha=0.5) + 
    ggtitle("Procruste rotation comparing NMDS from index to taxa; correlation = xx, p=xx") + scale_color_npg() + theme_bw()+ theme(panel.grid.major = element_blank(),panel.grid.minor = element_blank()) 


